### ISSUE TITLE
Form `isValid` not seeming to work

### ISSUE THREAD

**Issue by crutchcorn** [crutchcorn/houseform#47]
The following does not have a working `isValid` property

```jsx
import Head from 'next/head';
import styles from '../styles/Home.module.css';
import { Field, Form } from 'houseform';
import { z } from 'zod';

export default function Home() {
  return (
    <div className={styles.container}>
      <Head>
        <title>Create Next App</title>
      </Head>

      <main className={styles.main}>
        <Form
          onSubmit={(values) => {
            alert('Form was submitted with: ' + JSON.stringify(values));
          }}
        >
          {({ isValid, submit, errors }) => (
            <form
              onSubmit={(e) => {
                e.preventDefault();
                submit();
              }}
              className="flex flex-col items-center gap-6"
            >
              <Field
                name="email"
                onBlurValidate={z.string().email('Email not valid')}
              >
                {({ value, setValue, onBlur, errors }) => {
                  return (
                    <>
                      <input
                        className="daisy-input w-full max-w-[500px] rounded-full border border-dark-blue/[0.5] bg-beige"
                        placeholder="Email *"
                        type="email"
                        value={value}
                        onChange={(e) => setValue(e.target.value)}
                        onBlur={onBlur}
                      />
                      {errors.map((error) => (
                        <p key={error}>{error}</p>
                      ))}
                    </>
                  );
                }}
              </Field>
              <Field
                name="terms"
                initialValue={true}
                onChangeValidate={z.literal(true, {
                  errorMap: () => ({ message: 'Terms are required' }),
                })}
              >
                {({ value = false, setValue, onBlur, errors }) => {
                  return (
                    <>
                      <label className="flex items-center gap-2">
                        <input
                          className="daisy-checkbox"
                          type="checkbox"
                          checked={value}
                          onBlur={onBlur}
                          onChange={(e) => setValue(e.target.checked)}
                        />
                        <span>I accept the terms</span>
                      </label>
                      {errors.map((error) => (
                        <p key={error}>{error}</p>
                      ))}
                    </>
                  );
                }}
              </Field>
              <Field
                name="contact"
                initialValue={true}
                onChangeValidate={z.boolean()}
              >
                {({ value, setValue, onBlur, errors }) => {
                  return (
                    <>
                      <label className="flex items-center gap-2">
                        <input
                          className="daisy-checkbox"
                          type="checkbox"
                          checked={value}
                          onBlur={onBlur}
                          onChange={(e) => setValue(e.target.checked)}
                        />
                        <span>I want to be contacted (not required)</span>
                      </label>
                      {errors.map((error) => (
                        <p key={error}>{error}</p>
                      ))}
                    </>
                  );
                }}
              </Field>
              <p>{!errors?.length ? 'isvalid' : 'isNotValid'}</p>
              <button
                type="submit"
                disabled={!isValid}
                className="btn btn-icon btn-dark-blue disabled:bg-opacity-50"
              >
                Send
              </button>
            </form>
          )}
        </Form>
      </main>

      <footer className={styles.footer}>
        <a href="https://next.new" target="_blank" rel="noopener noreferrer">
          Created with&nbsp;<b>next.new</b>&nbsp;⚡️
        </a>
      </footer>
    </div>
  );
}
```

### Discussed in https://github.com/crutchcorn/houseform/discussions/46

<div type='discussions-op-text'>

<sup>Originally posted by **heggemsnes** March  5, 2023</sup>
Hi!

Testing out this library and it seems quite nice. 

I am a bit confused with the "isValid" boolean, I would think this would be false if there are any errors, but it seems it can be true. See an example using Next.js here: https://stackblitz.com/edit/nextjs-t6js87?file=pages/index.js

Basically just try to submit the form and "isValid" is still true.

In order to get the UX of a disabled button that is enabled when validated correctly I'm doing this now:

```typescript
<button
  type="submit"
  disabled={!isValid || errors.length > 0 || !isDirty}
  className="btn btn-icon btn-dark-blue disabled:bg-opacity-50"
>
  Send
</button>
```
Any pointers here would be appreciated!</div>

**Comment by crutchcorn** [comment]
This issue has now been fixed in `1.3.1`
